packages/wouter/types/index.d.ts

```diff
@@ -113,8 +107,18 @@ export function Redirect<H extends BaseLocationHook = BrowserLocationHook>(
   context?: any
 ): ReactElement<any, any> | null;
 
+type AsChildProps<RequiredProps, DefaultElementProps> =
```

Comment:
What do you think about renaming `RequiredProps` to something like `NavigateOptions`?

Reply:
This is a generic type that can be used with other components, not only with links.

I renamed this prop to `ComponentProps`, is this a better name?